src/riak_kv_pb_timeseries.erl

```diff
@@ -123,32 +126,35 @@ encode(Message) ->
 
 
 -spec process_tsreq(binary(), term(), #state{}) ->
-{reply, #tsqueryresp{} | #rpberrorresp{}, #state{}}.
+                           {reply, #tsqueryresp{} | #rpberrorresp{}, #state{}}.
 process_tsreq(Table, Data, State) ->
-  Mod = riak_ql_ddl:make_module_name(Table),
-
-  case (catch validate_rows(Mod, Data)) of
-    [] ->
-      try
-        case put_data(Data, Table, Mod) of
-          0 ->
-            {reply, #tsputresp{}, State};
-          ErrorCount ->
-            EPutMessage = flat_format("Failed to put ~b record(s)", [ErrorCount]),
-            {reply, make_rpberrresp(?E_PUT, EPutMessage), State}
-        end
-      catch
-        Class:Exception ->
-          lager:error("error: ~p:~p~n~p", [Class,Exception,erlang:get_stacktrace()]),
-          Error = make_rpberrresp(?E_IRREG, to_string({Class, Exception})),
-          {reply, Error, State}
-      end;
-    BadRowIdxs when is_list(BadRowIdxs) ->
-      {reply, validate_rows_error_response(BadRowIdxs), State};
-    {_, {undef, _}} ->
-      BucketProps = riak_core_bucket:get_bucket(table_to_bucket(Table)),
-      {reply, missing_helper_module(Table, BucketProps), State}
-  end.
+    check_table_and_call(
+      Table,
+      fun(Mod, _DDL) ->
```

Comment:
Maybe making this a named function might make the code more readable.  Same in the following cases.